src/main/java/edu/ksu/canvas/attendance/controller/SummaryController.java

```diff
@@ -78,47 +83,89 @@ private ModelAndView studentSummary(String sectionId, String studentId, boolean
-                reportService.getSimpleAttendanceSummaryReport(validatedSectionId) :
-                reportService.getAviationAttendanceSummaryReport(validatedSectionId);
-        List<LtiLaunchData.InstitutionRole> institutionRoles = canvasService.getRoles();
+        List<AttendanceStudent> studentAttendanceList = studentService.getStudentByCourseAndSisId(student.getSisUserId(), selectedCourseId);
 
-        student.getAttendances().sort(Comparator.comparing(Attendance::getDateOfClass).reversed());
+        List<AttendanceSection> sectionList = new ArrayList<>();
+        for (int i = 0; i < studentAttendanceList.size(); i++) {
+            sectionList.add(sectionService.getSection(studentAttendanceList.get(i).getCanvasSectionId()));
+        }
 
+        List<AttendanceSummaryModel> summaryForSections = new ArrayList<>();
         summaryForSections.stream()
                 .flatMap(summary -> summary.getEntries().stream())
-                .filter(entry -> entry.getStudentId() == validatedStudentId)
-                .findFirst()
-                .ifPresent(entry ->  page.addObject("attendanceSummaryEntry", courseConfigurationForm.getSimpleAttendance() ?
-                            new AttendanceSummaryModel.Entry(entry.getCourseId(), entry.getSectionId(), entry.getStudentId(), entry.getSisUserId(), entry.getStudentName(), student.getDeleted(), entry.getTotalClassesTardy(), entry.getTotalClassesMissed(), entry.getTotalClassesExcused(), entry.getTotalClassesPresent())
-                          : new AttendanceSummaryModel.Entry(entry.getCourseId(), entry.getSectionId(), entry.getStudentId(), entry.getSisUserId(), entry.getStudentName(), student.getDeleted(), entry.getSumMinutesMadeup(), entry.getRemainingMinutesMadeup(), entry.getSumMinutesMissed(), entry.getPercentCourseMissed())));
+                .filter(entry -> entry.getStudentId() == validatedStudentId);
+
+        summaryForSections.addAll(isSimpleAttendance ?
+                reportService.getSimpleAttendanceSummaryReport(student.getCanvasSectionId()) :
+                reportService.getAviationAttendanceSummaryReport(student.getCanvasSectionId()));
+
+        List<LtiLaunchData.InstitutionRole> institutionRoles = canvasService.getRoles();
 
         institutionRoles.stream()
                 .filter(institutionRole -> institutionRole.equals(LtiLaunchData.InstitutionRole.Learner))
                 .findFirst()
                 .ifPresent(role -> page.addObject("isStudent", true));
 
+        int totalTardy = 0, totalExcused = 0, totalMissed = 0;
+
+        if (!studentAttendanceList.isEmpty()) {
+            for (int i = 0; i < studentAttendanceList.size(); i++) {
+                AttendanceStudent current = studentAttendanceList.get(i);
+
+                for (int index = 0; index < current.getAttendances().size(); index++) {
+                    switch (current.getAttendances().get(index).getStatus()) {
```

Comment:
i think converting these two loops into something similar to the comment i left above would improve the readability of this code block and remove the need to use .get(index).getStatus/ .get(i);
also the if(!studentAttendanceList.isEmpty()) could be remove since the size would be zero and it would fail the initial condition check for the loop, i think.